frontends/web/src/routes/settings/manage-accounts.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2021 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h, RenderableProps } from 'preact';
+import { Link, route } from 'preact-router';
+import * as accountAPI from '../../api/account';
+import { getAccountFavorites, setAccountFavorite, TFavorites } from '../../api/config';
+import { Button, ButtonLink } from '../../components/forms';
+import Logo from '../../components/icon/logo';
+import { Entry } from '../../components/guide/entry';
+import { Guide } from '../../components/guide/guide';
+import { Header } from '../../components/layout';
+import { Toggle } from '../../components/toggle/toggle';
+import { translate, TranslateProps } from '../../decorators/translate';
+import * as style from './settings.css';
+
+interface SettingsProps {
+    accounts: accountAPI.IAccount[];
+    deviceIDs: string[];
+}
+
+type Props = SettingsProps & TranslateProps;
+
+interface State {
+    accountFavorites?: TFavorites;
+}
+
+class ManageAccounts extends Component<Props, State> {
+    public readonly state: State = {
+        accountFavorites: undefined,
+    };
+
+    public componentDidMount() {
+        getAccountFavorites()
+            .then(accountFavorites => {
+                this.setState({ accountFavorites });
+            })
+            .catch(console.error);
+    }
+
+    public componentDidUpdate(prevProps) {
+        if (prevProps.deviceIDs.length && !this.props.deviceIDs.length) {
+            route('/', true);
```

Comment:
I am afraid putting routing here instead of app.tsx (where deviceIDs come from and all other routing happens) makes routing harder to debug.

Besides, why do this route here based on devices anyway? This is an accounts page, not a devices page :)

Reply:
Yes completely agree, but copypasta from settings.tsx and there is some some logic that a home icon is shown when a device is plugged in.

Then since we don't have watch-only I wasn't sure if one should stay on the page when the device is unplugged.

Reply:
I'll remove the home icon as well.